cmd/podman/pod_stats.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"encoding/json"
+	tm "github.com/buger/goterm"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/cmd/podman/formats"
+	"github.com/projectatomic/libpod/cmd/podman/libpodruntime"
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/ulule/deepcopier"
+	"github.com/urfave/cli"
+)
+
+var (
+	podStatsFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "show stats for all pods.  Only running pods are shown by default.",
+		},
+		cli.BoolFlag{
+			Name:  "no-stream",
+			Usage: "disable streaming stats and only pull the first result, default setting is false",
+		},
+		cli.BoolFlag{
+			Name:  "no-reset",
+			Usage: "disable resetting the screen between intervals",
+		}, LatestFlag,
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "pretty-print container statistics to JSON or using a Go template",
+		},
+	}
+	podStatsDescription = "display a live stream of resource usage statistics for the containers in or more pods"
+	podStatsCommand     = cli.Command{
+		Name:                   "stats",
+		Usage:                  "Display percentage of CPU, memory, network I/O, block I/O and PIDs for containers in one or more pods",
+		Description:            podStatsDescription,
+		Flags:                  podStatsFlags,
+		Action:                 podStatsCmd,
+		ArgsUsage:              "[POD_NAME_OR_ID]",
+		UseShortOptionHandling: true,
+	}
+)
+
+func podStatsCmd(c *cli.Context) error {
+	var (
+		podFunc func() ([]*libpod.Pod, error)
+	)
+
+	format := c.String("format")
+	all := c.Bool("all")
+	latest := c.Bool("latest")
+	ctr := 0
+	if all {
+		ctr += 1
+	}
+	if latest {
+		ctr += 1
+	}
+	if len(c.Args()) > 0 {
+		ctr += 1
+	}
+
+	if ctr > 1 {
+		return errors.Errorf("--all, --latest and containers cannot be used together")
+	} else if ctr == 0 {
+		// If user didn't specify, imply --all
+		all = true
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	times := -1
+	if c.Bool("no-stream") {
+		times = 1
+	}
+
+	if len(c.Args()) > 0 {
+		podFunc = func() ([]*libpod.Pod, error) { return runtime.GetPodsByList(c.Args()) }
+	} else if c.IsSet("all") {
```

Comment:
Should this be the local variable 'all'?

Reply:
sure